pkg/ha/ha_tracker.go

```diff
@@ -182,6 +183,11 @@ func NewHATracker(cfg HATrackerConfig, limits HATrackerLimits, trackerStatusConf
 			Help: "The total number of CAS calls to the KV store for a user ID/cluster.",
 		}, []string{"user", "cluster"}),
 
+		userReplicaGroupCount: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "ha_tracker_user_replica_group_count",
+			Help: "Number of HA replica groups tracked for each user.",
+		}, []string{"user"}),
```

Comment:
We need to clean up the metric in `CleanupHATrackerMetricsForUser` so that when a user is not active anymore the corresponding user dimension is removed from the metric